django/db/backends/base/base.py

```diff
@@ -219,25 +242,43 @@ def get_new_connection(self, conn_params):
             "method"
         )
 
+    async def aget_new_connection(self, conn_params):
+        """Open a connection to the database."""
+        raise NotSupportedError(
+            "subclasses of BaseDatabaseWrapper may require an aget_new_connection() "
+            "method"
+        )
+
     def init_connection_state(self):
         """Initialize the database connection settings."""
         global RAN_DB_VERSION_CHECK
         if self.alias not in RAN_DB_VERSION_CHECK:
             self.check_database_version_supported()
             RAN_DB_VERSION_CHECK.add(self.alias)
 
+    async def ainit_connection_state(self):
+        """Initialize the database connection settings."""
+        global RAN_DB_VERSION_CHECK
+        if self.alias not in RAN_DB_VERSION_CHECK:
+            await self.acheck_database_version_supported()
+            RAN_DB_VERSION_CHECK.add(self.alias)
```

Comment:
Some people might see this and be worried, I _might_ have a good solution for this using `libcst` to be able to generate the original from the `a`'d implementation. Will try something out